_posts/2020-05-07-camel-kafka-connectors.md

```diff
@@ -0,0 +1,166 @@
+
+### Add the Camel Telegram Connector to the Kafka Connect image
+
+Download the [latest version](https://repo1.maven.org/maven2/org/apache/camel/kafkaconnector/camel-telegram-kafka-connector/0.1.0/camel-telegram-kafka-connector-0.1.0-package.tar.gz) of the Camel Telegram Connector from Maven Central (0.1.0 at the time of writing).
+
+Note: This will build a Kafka Connect image with just the Camel Telegram Connector in. It is also possible to [build an image](https://camel.apache.org/camel-kafka-connector/latest/try-it-out-on-openshift-with-strimzi.html) which contains all of the Camel Connectors. This involves building from the source and results in a larger image. However, it does have the benefit of being able to use a single Connect image to interact with multiple external systems.
+
+Add the connectors to the Kafka Connect by creating a new image from this dockerfile. If you are running on OpenShift you can use the [Connect S2I feature](https://camel.apache.org/camel-kafka-connector/latest/try-it-out-on-openshift-with-strimzi.html) instead of this step.
+
+```bash
+cat <<EOF >Dockerfile
+FROM strimzi/kafka:0.17.0-kafka-2.4.0
+USER root:root
+RUN mkdir -p /opt/kafka/plugins/camel
+COPY ./camel-telegram-kafka-connector-0.1.0-package.tar.gz /opt/kafka/plugins/camel/
+RUN tar -xvzf /opt/kafka/plugins/camel/camel-telegram-kafka-connector-0.1.0-package.tar.gz --directory /opt/kafka/plugins/camel
+RUN rm /opt/kafka/plugins/camel/camel-telegram-kafka-connector-0.1.0-package.tar.gz
+USER 1001
+EOF
+```
+
+Next, build the image. 
+
+```bash
+docker build . -t <docker-org>/camel-kafkaconnect
+docker push <docker-org>/camel-kafkaconnect
+```
+
+### Start a Kafka Connect Cluster with the Operator managing the Connectors. 
+
+Telegram requires a token to authenticate to the API. 
+Other connectors may require usernames and passwords or different types of keys. 
+It is possible to include these directly in the KafkaConnector custom resource. 
+However, that is clearly not very secure as any user who has read access to the custom resource could extract the credentials. 
+In order to secure the token we will add it to a Kubernetes `Secret` which will be mounted into the Connector pods. 
+
+Note: For instructions on how to create a bot and API key see the [Telegram documentation](https://core.telegram.org/bots).
+
+```bash
+cat <<EOF > telegram.properties
+token: <token>
+EOF
+kubectl -n kafka create secret generic telegram-credentials \
+  --from-file=telegram.properties
+```
+
+Strimzi provides two options for managing Kafka Connectors - either the Connect REST API or via an Operator. 
+We will use the latter so need to add the `strimzi.io/use-connector-resources: "true"` annotation to the `KafkaConnect` custom resource which enables this mode. 
+The secret will be loaded using Kafka's `config.providers` mechanism so set the `spec.config.config.providers.file.class` to the `FileConfigProvider`. 
```

Comment:
You could link to https://cwiki.apache.org/confluence/display/KAFKA/KIP-297%3A+Externalizing+Secrets+for+Connect+Configurations which explains config providers in much more detail.